### PR TITLE
Vim.Compat: Add several compatible functions

### PR DESCRIPTION
**Pull request by lambdalisue**
https://github.com/vim-jp/vital.vim/issues/343

### CODE REVIEW

File: doc/vital-vim-compat.txt [hunk 1]
```diff
@@ -34,6 +35,34 @@ has_version({version})			*Vital.Vim.Compat.has_version()*
...
 	Compatibility function of |shiftwidth()| available from Vim 7.3.694.
 
+writefile({list}, {fname}[, {flags}])	*Vital.Vim.Compat.writefile()*
+	Compatibility function of |writefile()| which supports 'a' flag from
+	Vim 7.4.503.
+
+redraw([{bang}])			*Vital.Vim.Compat.redraw()*
+	Compatibility function of |redraw| which has an issue on MacVim (GUI).
+	It calls |sleep| 1m just after |redraw| when called in MacVim (GUI),
+	otherwise it just calls |redraw|.
```
Comment (ujihisa) [inline doc/vital-vim-compat.txt]: `s/call/execute/g` because `redraw` is not a function but a command.

**Comment by lambdalisue** [comment]
MacVim's redraw/redrawstatus issue will be fixed https://github.com/macvim-dev/macvim/issues/174#issuecomment-164827421

I'm not sure if `Vim.Compat.redraw()` should exist or not now.

**Comment by rhysd** [comment]
Please read documentation.

> *Vital.Vim.Compat* provides vim compatibility wrapper functions for different
> versions/patchlevels of Vim.
>
> This module is not for multiple OS compatibilities but for versions of Vim
> itself.

`Vim.Compat` aims to absorb the difference among versions/patchlevels of Vim.  I think `redraw()` should be removed.

**Comment by lambdalisue** [comment]
Yes I first thought so as well but the word version can indicate "MacVim's specific version" and philosophy of the module stands for compatibility so I don't think the function against to the purpose of the module.

But as I mentioned, the feature is fixed today and (I think) people usually use HEAD version so, in this point of view, I'm not sure if the function should exist or force users to use a latest version of MacVim.

**Comment by lambdalisue** [comment]
Additional note:
redraw compat comes from a bug of (Mac)Vim, not because of OS difference. That why I think it should be in Vim.Compat if it is required.

**Comment by rhysd** [comment]
I agree that this is a problem of the term 'Vim compatibility'.

If the bug is included in 'compatibility' which this module must cover,  too many bugs should be covered by this module.  As you know, many bugs are reported and fixed by Vim developers.

I think this module should cover the API compatibility such as `glob()` and should not cover buggy behavior.  (Sure, I agree that it is useful to wrap `redraw()`.)

**Comment by lambdalisue** [comment]
> If the bug is included in 'compatibility' which this module must cover, too many bugs should be covered by this module. As you know, many bugs are reported and fixed by Vim developers.

Completely true. I changed my mind ;-)

**Comment by lambdalisue** [comment]
I just remove a workaround function for a `redraw` command while:

1. As @rhysd said, the module should **ONLY** stand for covering compatibility of difference of Vim's version and not for workarounds for known bugs.
2. `redraw` issue has been fixed from [this commit](https://github.com/macvim-dev/macvim/commit/e7ca3fe78949b61f1a4962075b638ceb46645195) on macvim-dev/macvim repository and the command `homebrew install --HEAD macvim`, which most of MacVim users use recently, installs the latest version.

Probably developers need to add `Vim.Compat.Workarounds.redraw()`, `Vim.MacVim.redraw()`, or whatever to add this function **if necessary**. For now, I don't think the workaround function is required anymore ;-)

**Comment by crazymaster** [comment]
>`https://github.com/vim-jp/vim/commit/8399b184df06f80ca030b505920dd3e97be72f20`

These links is not found.

**Comment by lambdalisue** [comment]
> These links is not found.

Hum... OK I'll remove these thanks

**Comment by ujihisa** [comment]
:+1: :sushi:

**Comment by rhysd** [comment]
@lambdalisue 

Thank you for understanding.  Looks good to me.  I think this PR is ready to merge after resolving conflict with `master` branch.

**Comment by ujihisa** [comment]
looks like it's mergeable

**Comment by rhysd** [comment]
@ujihisa 

Hmm,  at least in my browser it says 'This branch is out-of-date with the base branch'.

tmp

**Comment by ujihisa** [comment]
!?

**Comment by rhysd** [comment]
lol GitHub says @ujihisa should merge this.

**Comment by ujihisa** [comment]
:shipit: merge力 :shipit:

**Comment by rhysd** [comment]
:muscle::shipit:

**Comment by lambdalisue** [comment]
What lol. Anyway thanks ;-)

**Comment by rhysd** [comment]
Nice work, @lambdalisue !

**Pull request merged**